server/src/main/java/org/opensearch/index/store/remote/utils/FileType.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.store.remote.utils;
+
+import org.opensearch.common.annotation.ExperimentalApi;
+
+/**
+ * Enum to represent whether a file is block or not
+ *
+ * @opensearch.experimental
+ */
+@ExperimentalApi
+public enum FileType {
```

Comment:
This class feels a bit overkill for a single use in `listAll` method.  Why not match the regex there?